src/contracts/middleware/BLSPubkeyRegistry.sol

```diff
@@ -74,28 +74,32 @@ contract BLSPubkeyRegistry is IBLSPubkeyRegistry, Test {
     /**
      * @notice Deregisters the `operator`'s pubkey for the specified `quorumNumbers`.
      * @param operator The address of the operator to deregister.
+     * @param completeDeregistration Whether the operator is deregistering from all quorums or just some.
      * @param quorumNumbers The quourm numbers the operator is deregistering from, where each byte is an 8 bit integer quorumNumber.
      * @param pubkey The public key of the operator.
      * @dev access restricted to the RegistryCoordinator
-     * @dev Preconditions:
+     * @dev Preconditions (these are assumed, not validated in this contract):
      *         1) `quorumNumbers` has no duplicates
      *         2) `quorumNumbers.length` != 0
      *         3) `quorumNumbers` is ordered in ascending order
      *         4) the operator is not already deregistered
      *         5) `quorumNumbers` is the same as the parameter use when registering
      *         6) `pubkey` is the same as the parameter used when registering
      */   
-    function deregisterOperator(address operator, bytes memory quorumNumbers, BN254.G1Point memory pubkey) external onlyRegistryCoordinator returns(bytes32){
+    function deregisterOperator(address operator, bool completeDeregistration, bytes memory quorumNumbers, BN254.G1Point memory pubkey) external onlyRegistryCoordinator returns(bytes32){
         bytes32 pubkeyHash = BN254.hashG1Point(pubkey);
 
         require(pubkeyCompendium.pubkeyHashToOperator(pubkeyHash) == operator,"BLSPubkeyRegistry.registerOperator: operator does not own pubkey");
 
         // update each quorum's aggregate pubkey
-        _processQuorumApkUpdate(quorumNumbers, pubkey.negate());
-        // update the global aggregate pubkey
-        _processGlobalApkUpdate(pubkey.negate());
-        // emit event so offchain actors can update their state
-        emit PubkeyRemoved(operator, pubkey);
+        _processQuorumApkUpdate(operator, quorumNumbers, pubkey.negate());
+        
+        if(completeDeregistration){
+            // update the global aggregate pubkey
+            _processGlobalApkUpdate(pubkey.negate());
+            // emit event so offchain actors can update their state
+            emit PubkeyRemoved(operator, pubkey);
+        }
```

Comment:
what happens if someone deregisters from all the quorums but marks this as false? Seems like it could break properties we'd like to have, at least.

Reply:
oh nevermind, the function is permissioned, so it's just up to us to call it correctly then I guess.
is there a safer way of doing this then?

Reply:
we could specifically check with registry coordinator that the current quorumBitmap is the same as the quorumNumbers provided. would that make you more confident in this functionality for the cost? I'm fine either way.

Reply:
agreed